isis/tests/Fixtures.cpp

```diff
@@ -277,53 +277,22 @@ namespace Isis {
 
   void ApolloNetwork::SetUp() {
     TempTestingFiles::SetUp();
-    
-    isdFile1 = new FileName("data/apolloNetwork/apolloImage1.isd");
-    isdFile2 = new FileName("data/apolloNetwork/apolloImage2.isd");
-    isdFile3 = new FileName("data/apolloNetwork/apolloImage3.isd");
-    isdFile4 = new FileName("data/apolloNetwork/apolloImage4.isd");
-    isdFile5 = new FileName("data/apolloNetwork/apolloImage5.isd");
-    isdFile6 = new FileName("data/apolloNetwork/apolloImage6.isd");
-    isdFile7 = new FileName("data/apolloNetwork/apolloImage7.isd");
-
-    label1 = new FileName("data/apolloNetwork/apolloImage1.pvl");
-    label2 = new FileName("data/apolloNetwork/apolloImage2.pvl");
-    label3 = new FileName("data/apolloNetwork/apolloImage3.pvl");
-    label4 = new FileName("data/apolloNetwork/apolloImage4.pvl");
-    label5 = new FileName("data/apolloNetwork/apolloImage5.pvl");
-    label6 = new FileName("data/apolloNetwork/apolloImage6.pvl");
-    label7 = new FileName("data/apolloNetwork/apolloImage7.pvl");
-
-    cube1 = new Cube();
-    cube1->fromIsd(tempDir.path() + "/cube1.cub", *label1, *isdFile1, "rw");
-
-    cube2 = new Cube();
-    cube2->fromIsd(tempDir.path() + "/cube2.cub", *label2, *isdFile2, "rw");
-
-    cube3 = new Cube();
-    cube3->fromIsd(tempDir.path() + "/cube3.cub", *label3, *isdFile3, "rw");
-
-    cube4 = new Cube();
-    cube4->fromIsd(tempDir.path() + "/cube4.cub", *label4, *isdFile4, "rw");
 
-    cube5 = new Cube();
-    cube5->fromIsd(tempDir.path() + "/cube5.cub", *label5, *isdFile5, "rw");
-
-    cube6 = new Cube();
-    cube6->fromIsd(tempDir.path() + "/cube6.cub", *label6, *isdFile6, "rw");
-
-    cube7 = new Cube();
-    cube7->fromIsd(tempDir.path() + "/cube7.cub", *label7, *isdFile7, "rw");
+    isdFiles.fill(nullptr, 7);
+    labelFiles.fill(nullptr, 7);
+    cubes.fill(nullptr, 7);
 
     cubeList = new FileList();
-    cubeList->append(cube1->fileName());
-    cubeList->append(cube2->fileName());
-    cubeList->append(cube3->fileName());
-    cubeList->append(cube4->fileName());
-    cubeList->append(cube5->fileName());
-    cubeList->append(cube6->fileName());
-    cubeList->append(cube7->fileName());
-
+    
+    for(int i = 0; i < isdFiles.size(); i++) {
+      int n = i+1; // filenames use 1 based indexing 
+      isdFiles[i] = new FileName("data/apolloNetwork/apolloImage"+QString::number(n)+".isd");     
+      labelFiles[i] = new FileName("data/apolloNetwork/apolloImage"+QString::number(n)+".pvl");
+      cubes[i] = new Cube();
```

Comment:
Why not avoid the memory management issues and just start the vectors out empty and then fill them with the actual objects instead of just pointers?